source/common/config/well_known_names.cc

```diff
@@ -18,14 +20,18 @@ std::string expandRegex(const std::string& regex) {
               // underscores.
               {"<CIPHER>", R"([\w-]+)"},
               // A generic name can contain any character except dots.
-              {"<NAME>", R"([^\.]+)"},
+              {"<NAME>", NAME_REGEX},
               // Route names may contain dots in addition to alphanumerics and
               // dashes with underscores.
               {"<ROUTE_CONFIG_NAME>", R"([\w-\.]+)"}});
 }
 
 } // namespace
 
+const Regex::CompiledGoogleReMatcher& validTagValue() {
```

Comment:
expose a bool function rather than returning the complex interface which the caller then has to call. This also allows the freedom of changing the implementation without changing the uses.

You will still need a function that returns the interface as a private helper.

Reply:
> expose a bool function rather than returning the complex interface which the caller then has to call. This also allows the freedom of changing the implementation without changing the uses.

This is basically what I am doing but the function with the bool signature is in `tag_utility` header. I like it there more because it seems more intuitive but not a strong opinion

Reply:
That makes sense -- and you can have an interface in tag_utility.h, which can be an inline call to a bool method exposed here.

But exposing a `CompiledGoogleReMatcher&` here is a thicker coupling than necessary, IMO.